sourcetracker/_plot.py

```diff
@@ -23,3 +24,261 @@ def plot_heatmap(mpm, cm=plt.cm.viridis, xlabel='Sources', ylabel='Sinks',
     ax.set_ylabel(ylabel)
     ax.set_title(title)
     return fig, ax
+
+
+class ST_graphs:
+    def __init__(self, mpm, output_dir,
+                 title='Mixing Proportions', color='viridis'):
+        self.file = output_dir
+        self.mpm = mpm
+        self.title = title
+        self.color = color
+
+    def ST_heatmap(self, unknowns=True, annot=True,
+                   xlabel='Sources', ylabel='Sinks', vmax=1.0):
+        """
+        Standard heat map altered for custom
+        shape and direct png save function
+        """
+        prop = self.mpm
+        if not unknowns:
+            prop = prop.drop(['Unknown'], axis=1)
+            prop = prop.div(prop.sum(axis=1), axis=0)
+        fig, ax = plt.subplots(figsize=((prop.shape[1] + 6),
+                                        (prop.shape[0] * 3 / 4)+4))
+        sns.heatmap(prop, vmin=0.0, vmax=vmax, cmap=self.color,
+                    annot=annot, linewidths=.5, ax=ax)  # ,vmax=1.0
+        ax.set_xlabel(xlabel)
+        ax.set_ylabel(ylabel)
+        ax.set_title(self.title)
+        plt.xticks(rotation=45, ha='right')
+        if unknowns:
+            plt.savefig(os.path.join(self.file,
+                                     self.title + "_heatmap.png"))
+        else:
+            plt.savefig(os.path.join(self.file,
+                                     self.title + "_heatmap_nounknown.png"))
```

Comment:
When we meet can you explain the reason that you are checking for whether there are unknowns? I dont fully understand